gateway/modules/schema/creation_test.go

```diff
@@ -1557,6 +1557,19 @@ func TestSchema_generateCreationQueries(t *testing.T) {
 			want:    []string{"CREATE TABLE test.table1 (col2 timestamp );", "ALTER TABLE test.table1 ADD CONSTRAINT c_table1_col2 FOREIGN KEY (col2) REFERENCES test.table2 (id)"},
 			wantErr: false,
 		},
+		{
+			name: "adding a table and a foreign key with ON CASCADE DELETE",
```

Comment:
Add the following test cases
- add a table along with foreign key with on cascade delete (you already have this
- Normal foreign key to foreign key with cascade delete
- Opposite of the above